fix: ensure message content is not modified by header filter

georglauterbach opened the pull request:
# Description

<!--
  Include a summary of the change.
  Please also include relevant motivation and context.
-->

**I was unable to extend the set of changes from #4425. Hence, it was quicker for me to simply open a new PR.**

CC @mpldr 

---

Replaces #4425. This MR contains a superset of changes to `sender_header_filter.pcre`, also cleaning the file up a bit (sorting it alphabetically).

<!-- Link the issue which will be fixed (if any) here: -->
Fixes issue with GPG, see linked PR.

## Type of change

<!-- Delete options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary, I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] **I have added information about changes made in this PR to `CHANGELOG.md`**

mpldr commented:
I've integrated the changes into #4425 and updated it accordingly. I've removed the other header filters, though.

georglauterbach commented:
> I've integrated the changes into #4425 and updated it accordingly. I've removed the other header filters, though.

I do not see the changes yet. Did you push them already?

And what do you mean by "removed the other header filters"? I'd rather we leave the other filters as is.

mpldr commented:
Done in #4429
